### PR TITLE
Remove Scope

### PR DESCRIPTION
**Pull request by Rachelwrq**
For issue #328 deprecated `Scope` class to fit OpenAPI 3.0, now `scope` is just a simple **Map** of strings.

### CODE REVIEW

**Comment by genie-microprofile** [comment]
Can one of the admins verify this patch?

**Comment by jmini** [comment]
@Rachelwrq thank you for this patch.

I will review more in depth, but I have the feeling that with the way this PR is written, `Scopes` can't be used anymore. Existing code will break (`Scopes` could be completely removed, it is no longer used in the model at all) and there was no previous released version indicating that this will be the case.

---

For the moment all the breaking changes introduced with `2.0` are marked as deprecated in `1.1`. If you remove all those warnings with `1.1` (and switch to the new way of expressing stuff) and then switch to `2.0` it will work.

I was thinking that this is the rule in Microprofile projects.

--- 
As a reference have a look on how @EricWittmann prepared the removal of `ServerVariables` in `1.1`:
* b459d3d20ddb60d2c009727769568831ff1e6f18
* f7970b9d0787bbd7f70f06fa7a52973b16ca1f33
* 2dbe14058058bbd831dba8a24dd48d0d8e07c727

**Comment by arthurdm** [comment]
hi @jmini - I am wondering if since now master == `2.0`,  would we continue to update the `1.x` branch?

**Comment by Rachelwrq** [comment]
@jmini Thank you for looking into my PR.

I made some modifications in reference to Eric's commits as well as the most recent files relating to `ServerVariables`; so that the `Scopes` is properly deprecated by now. 

Please let me know if there is anything else I might have missed.

**Comment by jmini** [comment]
> I am wondering if since now master == 2.0, would we continue to update the 1.x branch?

@arthurdm I do not think so.

In my opinion, `1.x` is the maintenance branch for the `1.1.x` versions (if something is found that requires to build again a `1.1.x` release)

**Comment by jmini** [comment]
@Rachelwrq I will. 

I would like to test it with my implementation, but I am a little behind with catching up with the changes on master. I hope you do not mind the delay...

**Comment by arthurdm** [comment]
> In my opinion, 1.x is the maintenance branch for the 1.1.x versions (if something is found that requires to build again a 1.1.x release)

So with that thought, do we really need to deprecate things?  I was thinking, until `2.0` is not released, we can do breaking changes in master?

**Comment by jmini** [comment]
> we can do breaking changes in master?

You are the MicroProfile expert, I have no idea of the rules... Should I ask the root project for opinion?

In my eyes there are 2 types of breaking changes (I just made up the names, maybe there is an official terminology). Applied to removal:

* A) Hard breaking change: you just do not care about backward compatibility. You are allowed to remove what you want when you introduce a new major release.
* B) Soft breaking change: you are only allowed to remove something in a major version that was marked as deprecated in a previous release (and some projects add a notion of time: the item being removed must be marked as deprecated during X releases or during X months).

Java SE is clearly type B. The eclipse platform (Eclipse RPC, IDE - which has nothing to do with MicroProfile) is also type B.

If you tell me than type A is allowed here, it makes our life easier!

**Comment by pilhuhn** [comment]
@arthurdm In Metrics we have iirc deprecated one thing in 1.x (if at all) and are making a lot of breaking changed in MP-Metrics 2.0. Those breaking changes are the reason we did not hurry at the end to get into MicroProfile 2.2, as those minor releases of the umbrella are not meant to have breaking changes. 
tl;dr: I'd say as long as you release a 2.0 of OpenAPI and 2.0 goes into MP-3.0, you should be fine.

**Comment by arthurdm** [comment]
thanks @pilhuhn!

**Comment by arthurdm** [comment]
hey @Rachelwrq - sorry for taking long to circle around this issue.  We had a chat at the last hangout and decided that:

- this PR, as it targets the `2.0` branch, is allowed to break the API and remove interfaces/methods
- a new PR should be created, to target the `1.x` branch, where you have deprecations in a non-breaking way.

**Comment by Rachelwrq** [comment]
Hi @arthurdm 
I’m just wondering if it is okay for me to just edit this PR so it is pushed to the `1.x` branch since I made all my changes to properly deprecate `Scope` object. I can then introduce breaking changes and delete interface and method in another PR.

**Comment by jmini** [comment]
@arthurdm: Thank you a lot for summarizing the hangout discussion. I like the outcome.

**Comment by jmini** [comment]
I do not remember where (Gitter?), but we were told to have an document or a section listing all the changes with the next version. This comment https://github.com/eclipse/microprofile-open-api/issues/251#issuecomment-481403906 is also mentioning it, but it does not tell where to put this.

**Comment by arthurdm** [comment]
@jmini - I believe you're looking for something [like this](https://github.com/eclipse/microprofile-config/blob/master/spec/src/main/asciidoc/release_notes.asciidoc)?    definitely a good point.

**Comment by jmini** [comment]
@arthurdm this means we do not have something like that in Microprofile-OpenAPI ?

**Comment by arthurdm** [comment]
My mistake, I used the wrong link.  We do have it [here](https://github.com/eclipse/microprofile-open-api/blob/master/spec/src/main/asciidoc/release_notes.asciidoc), but as per https://github.com/eclipse/microprofile-open-api/issues/343#issuecomment-476318978 we missed the update in `1.1`

**Comment by arthurdm** [comment]
added to the list.  thanks!
